README.md
# R-practicles
R for Data science


R Practical/history
mean(range(1,1000,2))
mean(range(1,10000,2))
mean(range(1,10000,12325))
gl(3,1000)
gl(4,3)
source('~/.active-rstudio-document', echo=TRUE)
gl(4,1/0)
levels(lv)
levels(lv)
lv < - c(T,F,T)
levels(lv)
lv <- c(T,F,T)
levels(lv)
is,logical(lv)
is.logical(lv)
x <- 0:1000
sum(x)
multipleClasses(x)
sum(x[x<595])
sub(x)
sub(x,0)
sub(x,x)
sub(1000,x)
sub(10x)
sub(10:x)
sub(10:x:9)
attach(data)
names(data)
attach(data)
names(data)
tapply(100, 23, 2x+45y+56z-67f)
counts <- rnbinom(10000,mu=0.92,size = 1.1)
counts[1:30]
table(counts)
View(bacteria)
q()
x <- 0:10
sum(x<5)
sum(x>5)
sum(x[x<5])
x<5
x>5
x<=5
1*(x<5)
x*(x<5)
x*(x>10)
x*(x<10)
x*(x<11)
x*(x<1/0)
x*(x<(1/0))
x <- 0:1/0
x*(x<(1/0))
x*(x<5)
1*(x<5)
x>5
x<5
sum(x[x<5])
sum(x>5)
x <- 0:(1/0)
x <- 0:100:10
x <- 0:90:10
x <- 0:90:1
x <- 0:90:0
x <- 0:90
x*(x<(1/0))
x*(x<(1/0))
x <- 0:90000
x*(x<(1/0))
sum(x[x<5])
sum(x>5)
x*(x<5)
x*(x>5)
x*(x<(1/0))
hist(x)
hist(x>889)
hist(x>88)
hist(x=90)
hist(x)
a <- rnorm(10,25,20)
View(draw.plotmath.cell)
xv <- rnorm(1000,100,10)
which(abs(xv-108)==min(abs(xv-108)))
which(abs(xv-108)==min(abs(xv-108)))
xv <- rnorm(1000,100,10)
xv <- rnorm(1000,100,10)
which(abs(xv-108)==min(abs(xv-108)))
xv <- rnorm(1000,100,10)
which(abs(xv-108)==min(abs(xv-108)))
x*(x>5)
sum(x*(x<5))
sum(x*(x>5))
y <- c(8,3,45,67,32143,212,45,58,2,3,5,7,8,6,0)
sort(y)
rev(sort(y))
sum(rev(sort(y))[1:3])
sum(rev(sort(y))[1:0])
sum(rev(sort(y))[1:3])
which(x==max(x))
which(x==min(x))
which.max(x)
which.min(x)
mean(y,x)
mean(y)
cov(x,y)
x <- 0:90000
y <- c(8,3,45,67,32143,212,45,58,2,3,5,7,8,6,0)
cov(x,y)
y <- 0:1000
cov(x,y)
x <- 0:90000
sum(rev(sort(y))[1:3])
y <- 0:1000
x <- 0:90000
cov(x,y,7)
cov(x,y)
cov(x,y,0)
cov(x,9,0)
cov(3,9,0)
cov(7,8)
cov(x,y)
cov(345432,5236345)
source('~/.active-rstudio-document', echo=TRUE)
cov(0,111)
/sdakm
//fefm
cov(x,65)
cov(3:32,65)
range(y)
fivenum(y)
counts <- rnbinom(10000,mu=0.92,size = 1.1)
counts[1:1000]
counts[1:1000]table(counts)
table(counts)
library(survey)
View(make.table)
View(oldpar)
View(gapminder)
counts
table(counts)
counts <- rnbinom(10000,mu=0.92,size = 1.1)
counts[1:1000]
table(counts)
counts
table(counts)
tapply(temperature, yr, mean,na.rm=TRUE)
poisson(0.95)
(poisson <- rpois(150,0.7))
rle(poisson)
max(rle(poisson))
max(rle(poisson))[[1]]
max(rle(poisson)[[1]])
max(rle(poisson)[[1]]==7)
which(rle(poisson)[[1]]==7)
which(rle(poisson)[[1]]==7)
rle(poisson)[[2]][[5]]
rle(poisson)[[2]][[5]]
rle(poisson)[[2]][[5]]
rle(poisson)[[2]][[98]]
rle(poisson)[[2]][[1]]
run.and.value(poisson)
run,and.value(poisson)
run.and.value(poisson)
run.and.value <- function(x){
a <- max(rle(poisson)[[1]])
b <- rle(posisson)[[2]][which(rle(poisson)[[1]] == a)]
cat("length = ",a,"value = ",b"\n")}
run.and.value(poisson)
length = 7 value = 0
length(rle(poisson)[[2]])
length(rle(poisson)[[2]])
(poisson <- rpois(150,0.7))
counts <- rnbinom(10000,mu=0.92,size = 1.1)
(poisson <- rpois(150,0.7))
rle(poisson)
which(rle(poisson)[[1]]==7)
rle(poisson)[[2]][[1]]
run.and.value <- function(x){
a <- max(rle(poisson)[[1]])
b <- rle(posisson)[[2]][which(rle(poisson)[[1]] == a)]
cat("length = ",a,"value = ",b"\n")}
run.and.value(poisson)
length = 7 value = 0
length(rle(poisson)[[2]])
source('~/.active-rstudio-document', echo=TRUE)
install.packages("PACKAGE:DATA SETS")
install.packages("learnr")
install.packages("PACKAGE:DATA SETS")
library(ggplot2)
install.packages('ggplot2')
install.packages('ggplot2')
library(ggplot2)
install.packages('ggplot2')
mpg
X = LOG(2)
X = C(10,20,30,40)
X = C(10,20,30,40)
x = c(10,20,30,40)
x
mean(x)
median(x)
mode(x)
# to enter data in r console in vector form (single var)
#to create a numeric object (or variable
y = c(11,12,13)
z = c(x,y)
mean(z)
cov(z)
co varince(z)
# to enter text or string object
A = C("section A","section b")
A = c("dsa","dawfa")
A
# x is a numeric object & A is text .Then  combine (x & A )
B = c(x,A)
B
#To  enter patterd  data
# TO ENTER PATTERNED DATA
# command names :"rep()"& "seq()"
x = rep(2,7)
# rep(vslue, no of times)
x
y = rep(c(1,2),3)
y
z =rep(c(1,2),c(3,3))
z
f = rep(c(1,2,3),c(1,2,3))
f
z = rep(c(1:100),c(1:1))
z
A = rep("low","high"),c(2,2))
A
A = rep(c("low","high"),c(2,2))
A
# seq (from value, to value, by differnce
a = seq(1,20,5)
a
a = seq(1,100,4)
a
a = seq(1,20,2)
a
b = rep(seq(2,20,4),2)
b
b = seq(rep(seq(2,20,4),c(1:5)),100,2)
b = seq(rep(seq(2,20,4),c(1:5)),20,2)
b = seq(rep(seq(2,20,4),c(1:5)),10,2)
b = rep(rep(seq(2,20,4),c(1:5)),100)
b
q()
x=seq(12:134:4)
x=seq(1,23,11)
x
x=rep(c(1:100),c(2:4)
x=rep(c(1:100),c(2:4)
x=rep(c(1:100),c(2:4)
x=rep(c(1:100),c(2:4)
x=rep(c(1:100),c(2:4)
x=rep(c(1:100),c(2:4)
x
x=rep(c(1:100),c(2:4))
x=rep(c(1:100),2)
x
x=seq(c(1:56),3,2)
x=seq(1,34,6)
y=seq(1,90,35
z=xy
y=seq(1.90,35)
z=xy
z=c(x,y)
mean(z)
median(z)
mode(z)
z
z=c(xy)
z=c(xyz)
f=c(x,y,z)
mean(f,z)
var(z)
var(x,y,z,f)
var(z,y,f)
var(z,f)
var(f)
cov
mean
cov
cov(x,y)
covarince(x,y)
cov(x)
cov(xy)
cov(x,f)
mode
median
rep
seq
int
integer
integer(x,y,z,f)
integer(x)
mod
mod
Mod
Arg
Conj
TRUE
FALSE
Mod(z)
Arg(z)
Arg(z,f)
MOd( z,f)
Mod(z,f)
as.complex(z)
exp(-2)
exp(-1)
exp(-1/0)
log(o)
log(0)
log(-0)
log(1/0)
log(pi/2)
log(cos(sin(pi/35)))
sininv(1)
sinh(1)
sin(1)
q()
q()
x=seq(c(1:56),3,2)
x=seq(c(1,3,2)
x
x
x=seq(c(1,3,2))
x
\\
/\
/\3
3#
23
34+
2432+$2
24+42
x=(1,2,3,45,6,78)
x=c(1,2,3,45,6,78)
y=c("abhishek kumar","open","closed","open","closed")
file1=list(Number=x,Area=y)
file1
list
#to check nature
str(file1
str(file1)
str(file1)
#Data frame
data1=data.frame(x,y)
y=legth(5)
y=length(5)
y
data1=data.frame(x,y)
data1
colnames(data1)c("Number","Area")
colnames(data1)=c("Number","Area")

R Practical/pr11.R
practical: 11

Q1) suppose a manufacturing operation produces items in hourly basis with 20%
    defective items per hour.
a) simulate the number of defectives for each hour of a 24-hour period,assuming 
    25 units are being produced per hour.
b)check if the number of defectives ever exceeds 5 from the simulated sample.
   compare this with the theoretical probability that the number of defectives 
   would exceed 5.
c)what is the probability that the number of defectives from the process
  will be atleast 3 and and atmost 7?
d)what is the the 70th and 90th quantiles for number of defectives?

x:binom rv of total no of defectives in 25 trials 
p= 0.2
#part a
x=rbinom(24,25,p)
x
#(b): Number of defectives ever exceeding 5 from the sample polpulation
y=numeric()
for(i in 1:24){
	if(x[i]>5){y[i]=1
	}else{y[i]=0}
}
y
sample.prop=mean(y)

prob.defect.more5=1-pbinom(5,25,p)   #p(x>5)1-p(x<=5)
c(sample.prop,prob.defect.more5)
	
#(c):probability that no of defectives will be at least 3&7 or binomial ditribution
prob.defect.atleast3=1-pbinom(2,25,p)
prob.defect.atmost7=pbinom(7,25,p)
c(prob.defect.atleast3,prob.defect.atmost7)

quantile.prob70=qbinom(0.70,25,p)
quantile.prob90=qbinom(0.90,25,p)
c(quantile.prob70,quantile.prob90)

R Practical/multiple lines.R
##plot(x, ....) :line graphs for single var
##plot(x,y,......):scatter plot
plot(control$Average.Diameter,xlab='Subgroups',ylab='Sample Mean',main='control chart')

  type arguments within plot(): by default 'p'; others 'l', 'c' ,'o','b'
plot(control$Average.Diameter,type='o',xlab='Subgroups',ylab='Sample Mean',main='control chart')
plot(control$Average.Diameter,type='c',xlab='Subgroups',ylab='Sample Mean',main='control chart')
plot(control$Average.Diameter,type='b',xlab='Subgroups',ylab='Sample Mean',main='control chart')
plot(control$Average.Diameter,type='l',xlab='Subgroups',ylab='Sample Mean',main='control chart')
plot(control$Average.Diameter,type='o',lty=3,lwd=2,pch=4,xlab='Subgroups',ylab='Sample Mean',main='control chart')
plot(control$Average.Diameter,type='o',lty=3,lwd=2,pch=4,xlab="",ylab="")
title(xlab='Subgroups',ylab='Sample Mean',main="control chart")
## multiple line graphs
plot(control$Average.Diameter,type='o',lty=1,lwd=2,pch=14,xlab='Subgroups',ylab='Sample Mean',main='control chart')
lines(control$UCLm,type='l',lty=2,lwd=2)
lines(control$CLm,type='l',lty=3,lwd=2)
lines(control$LCLm,type='l',lty=4,lwd=2)
legend('topleft',legend=c('Mean','UCL','CL','LCL'),lty=c(1:4),bty='n')
plot(control$Average.Diameter,control$SD.,pch=16)

R Practical/practical4R.R
>#PRACTICAL 4 
> faithful
    eruptions waiting
1       3.600      79
2       1.800      54
3       3.333      74
4       2.283      62
5       4.533      85
6       2.883      55
7       4.700      88
8       3.600      85
9       1.950      51
10      4.350      85
11      1.833      54
12      3.917      84
13      4.200      78
14      1.750      47
15      4.700      83
16      2.167      52
17      1.750      62
18      4.800      84
19      1.600      52
20      4.250      79
21      1.800      51
22      1.750      47
23      3.450      78
24      3.067      69
25      4.533      74
26      3.600      83
27      1.967      55
28      4.083      76
29      3.850      78
30      4.433      79
31      4.300      73
32      4.467      77
33      3.367      66
34      4.033      80
35      3.833      74
36      2.017      52
37      1.867      48
38      4.833      80
39      1.833      59
40      4.783      90
41      4.350      80
42      1.883      58
43      4.567      84
44      1.750      58
45      4.533      73
46      3.317      83
47      3.833      64
48      2.100      53
49      4.633      82
50      2.000      59
51      4.800      75
52      4.716      90
53      1.833      54
54      4.833      80
55      1.733      54
56      4.883      83
57      3.717      71
58      1.667      64
59      4.567      77
60      4.317      81
61      2.233      59
62      4.500      84
63      1.750      48
64      4.800      82
65      1.817      60
66      4.400      92
67      4.167      78
68      4.700      78
69      2.067      65
70      4.700      73
71      4.033      82
72      1.967      56
73      4.500      79
74      4.000      71
75      1.983      62
76      5.067      76
77      2.017      60
78      4.567      78
79      3.883      76
80      3.600      83
81      4.133      75
82      4.333      82
83      4.100      70
84      2.633      65
85      4.067      73
86      4.933      88
87      3.950      76
88      4.517      80
89      2.167      48
90      4.000      86
91      2.200      60
92      4.333      90
93      1.867      50
94      4.817      78
95      1.833      63
96      4.300      72
97      4.667      84
98      3.750      75
99      1.867      51
100     4.900      82
101     2.483      62
102     4.367      88
103     2.100      49
104     4.500      83
105     4.050      81
106     1.867      47
107     4.700      84
108     1.783      52
109     4.850      86
110     3.683      81
111     4.733      75
112     2.300      59
113     4.900      89
114     4.417      79
115     1.700      59
116     4.633      81
117     2.317      50
118     4.600      85
119     1.817      59
120     4.417      87
121     2.617      53
122     4.067      69
123     4.250      77
124     1.967      56
125     4.600      88
126     3.767      81
127     1.917      45
128     4.500      82
129     2.267      55
130     4.650      90
131     1.867      45
132     4.167      83
133     2.800      56
134     4.333      89
135     1.833      46
136     4.383      82
137     1.883      51
138     4.933      86
139     2.033      53
140     3.733      79
141     4.233      81
142     2.233      60
143     4.533      82
144     4.817      77
145     4.333      76
146     1.983      59
147     4.633      80
148     2.017      49
149     5.100      96
150     1.800      53
151     5.033      77
152     4.000      77
153     2.400      65
154     4.600      81
155     3.567      71
156     4.000      70
157     4.500      81
158     4.083      93
159     1.800      53
160     3.967      89
161     2.200      45
162     4.150      86
163     2.000      58
164     3.833      78
165     3.500      66
166     4.583      76
167     2.367      63
168     5.000      88
169     1.933      52
170     4.617      93
171     1.917      49
172     2.083      57
173     4.583      77
174     3.333      68
175     4.167      81
176     4.333      81
177     4.500      73
178     2.417      50
179     4.000      85
180     4.167      74
181     1.883      55
182     4.583      77
183     4.250      83
184     3.767      83
185     2.033      51
186     4.433      78
187     4.083      84
188     1.833      46
189     4.417      83
190     2.183      55
191     4.800      81
192     1.833      57
193     4.800      76
194     4.100      84
195     3.966      77
196     4.233      81
197     3.500      87
198     4.366      77
199     2.250      51
200     4.667      78
201     2.100      60
202     4.350      82
203     4.133      91
204     1.867      53
205     4.600      78
206     1.783      46
207     4.367      77
208     3.850      84
209     1.933      49
210     4.500      83
211     2.383      71
212     4.700      80
213     1.867      49
214     3.833      75
215     3.417      64
216     4.233      76
217     2.400      53
218     4.800      94
219     2.000      55
220     4.150      76
221     1.867      50
222     4.267      82
223     1.750      54
224     4.483      75
225     4.000      78
226     4.117      79
227     4.083      78
228     4.267      78
229     3.917      70
230     4.550      79
231     4.083      70
232     2.417      54
233     4.183      86
234     2.217      50
235     4.450      90
236     1.883      54
237     1.850      54
238     4.283      77
239     3.950      79
240     2.333      64
241     4.150      75
242     2.350      47
243     4.933      86
244     2.900      63
245     4.583      85
246     3.833      82
247     2.083      57
248     4.367      82
249     2.133      67
250     4.350      74
251     2.200      54
252     4.450      83
253     3.567      73
254     4.500      73
255     4.150      88
256     3.817      80
257     3.917      71
258     4.450      83
259     2.000      56
260     4.283      79
261     4.767      78
262     4.533      84
263     1.850      58
264     4.250      83
265     1.983      43
266     2.250      60
267     4.750      75
268     4.117      81
269     2.150      46
270     4.417      90
271     1.817      46
272     4.467      74
> head(faithful)
  eruptions waiting
1     3.600      79
2     1.800      54
3     3.333      74
4     2.283      62
5     4.533      85
6     2.883      55
> tail(faithful)
    eruptions waiting
267     4.750      75
268     4.117      81
269     2.150      46
270     4.417      90
271     1.817      46
272     4.467      74
> str(faithful)
'data.frame':   272 obs. of  2 variables:
 $ eruptions: num  3.6 1.8 3.33 2.28 4.53 ...
 $ waiting  : num  79 54 74 62 85 55 88 85 51 85 ...
> #continous Frequency Dist Table
> range(faithful$eruptions)
[1] 1.6 5.1
> #Define Class Interval
> breaks=seq(1.5,5.5,by=0.5)
> braks
Error: object 'braks' not found
> breaks
[1] 1.5 2.0 2.5 3.0 3.5 4.0 4.5 5.0 5.5
> class.int=cut(faithful$eruptions,breaks,right=F)
> clas.int
Error: object 'clas.int' not found
> class.int
  [1] [3.5,4) [1.5,2) [3,3.5) [2,2.5) [4.5,5) [2.5,3) [4.5,5) [3.5,4) [1.5,2) [4,4.5) [1.5,2) [3.5,4) [4,4.5) [1.5,2) [4.5,5) [2,2.5) [1.5,2) [4.5,5) [1.5,2) [4,4.5)
 [21] [1.5,2) [1.5,2) [3,3.5) [3,3.5) [4.5,5) [3.5,4) [1.5,2) [4,4.5) [3.5,4) [4,4.5) [4,4.5) [4,4.5) [3,3.5) [4,4.5) [3.5,4) [2,2.5) [1.5,2) [4.5,5) [1.5,2) [4.5,5)
 [41] [4,4.5) [1.5,2) [4.5,5) [1.5,2) [4.5,5) [3,3.5) [3.5,4) [2,2.5) [4.5,5) [2,2.5) [4.5,5) [4.5,5) [1.5,2) [4.5,5) [1.5,2) [4.5,5) [3.5,4) [1.5,2) [4.5,5) [4,4.5)
 [61] [2,2.5) [4.5,5) [1.5,2) [4.5,5) [1.5,2) [4,4.5) [4,4.5) [4.5,5) [2,2.5) [4.5,5) [4,4.5) [1.5,2) [4.5,5) [4,4.5) [1.5,2) [5,5.5) [2,2.5) [4.5,5) [3.5,4) [3.5,4)
 [81] [4,4.5) [4,4.5) [4,4.5) [2.5,3) [4,4.5) [4.5,5) [3.5,4) [4.5,5) [2,2.5) [4,4.5) [2,2.5) [4,4.5) [1.5,2) [4.5,5) [1.5,2) [4,4.5) [4.5,5) [3.5,4) [1.5,2) [4.5,5)
[101] [2,2.5) [4,4.5) [2,2.5) [4.5,5) [4,4.5) [1.5,2) [4.5,5) [1.5,2) [4.5,5) [3.5,4) [4.5,5) [2,2.5) [4.5,5) [4,4.5) [1.5,2) [4.5,5) [2,2.5) [4.5,5) [1.5,2) [4,4.5)
[121] [2.5,3) [4,4.5) [4,4.5) [1.5,2) [4.5,5) [3.5,4) [1.5,2) [4.5,5) [2,2.5) [4.5,5) [1.5,2) [4,4.5) [2.5,3) [4,4.5) [1.5,2) [4,4.5) [1.5,2) [4.5,5) [2,2.5) [3.5,4)
[141] [4,4.5) [2,2.5) [4.5,5) [4.5,5) [4,4.5) [1.5,2) [4.5,5) [2,2.5) [5,5.5) [1.5,2) [5,5.5) [4,4.5) [2,2.5) [4.5,5) [3.5,4) [4,4.5) [4.5,5) [4,4.5) [1.5,2) [3.5,4)
[161] [2,2.5) [4,4.5) [2,2.5) [3.5,4) [3.5,4) [4.5,5) [2,2.5) [5,5.5) [1.5,2) [4.5,5) [1.5,2) [2,2.5) [4.5,5) [3,3.5) [4,4.5) [4,4.5) [4.5,5) [2,2.5) [4,4.5) [4,4.5)
[181] [1.5,2) [4.5,5) [4,4.5) [3.5,4) [2,2.5) [4,4.5) [4,4.5) [1.5,2) [4,4.5) [2,2.5) [4.5,5) [1.5,2) [4.5,5) [4,4.5) [3.5,4) [4,4.5) [3.5,4) [4,4.5) [2,2.5) [4.5,5)
[201] [2,2.5) [4,4.5) [4,4.5) [1.5,2) [4.5,5) [1.5,2) [4,4.5) [3.5,4) [1.5,2) [4.5,5) [2,2.5) [4.5,5) [1.5,2) [3.5,4) [3,3.5) [4,4.5) [2,2.5) [4.5,5) [2,2.5) [4,4.5)
[221] [1.5,2) [4,4.5) [1.5,2) [4,4.5) [4,4.5) [4,4.5) [4,4.5) [4,4.5) [3.5,4) [4.5,5) [4,4.5) [2,2.5) [4,4.5) [2,2.5) [4,4.5) [1.5,2) [1.5,2) [4,4.5) [3.5,4) [2,2.5)
[241] [4,4.5) [2,2.5) [4.5,5) [2.5,3) [4.5,5) [3.5,4) [2,2.5) [4,4.5) [2,2.5) [4,4.5) [2,2.5) [4,4.5) [3.5,4) [4.5,5) [4,4.5) [3.5,4) [3.5,4) [4,4.5) [2,2.5) [4,4.5)
[261] [4.5,5) [4.5,5) [1.5,2) [4,4.5) [1.5,2) [2,2.5) [4.5,5) [4,4.5) [2,2.5) [4,4.5) [1.5,2) [4,4.5)
Levels: [1.5,2) [2,2.5) [2.5,3) [3,3.5) [3.5,4) [4,4.5) [4.5,5) [5,5.5)
> frequency=table(class.int)
> frequency
class.int
[1.5,2) [2,2.5) [2.5,3) [3,3.5) [3.5,4) [4,4.5) [4.5,5) [5,5.5) 
     51      41       5       7      30      73      61       4 
> relative.freq=frequency/272
> relative.freq
class.int
   [1.5,2)    [2,2.5)    [2.5,3)    [3,3.5)    [3.5,4)    [4,4.5)    [4.5,5)    [5,5.5) 
0.18750000 0.15073529 0.01838235 0.02573529 0.11029412 0.26838235 0.22426471 0.01470588 
> 